components/power-supply/component.go

```diff
@@ -79,7 +88,49 @@ func (c *component) States(ctx context.Context) ([]components.State, error) {
 }
 
 func (c *component) Events(ctx context.Context, since time.Time) ([]components.Event, error) {
-	return nil, nil
+	if runtime.GOOS != "linux" {
+		return nil, nil
+	}
+
+	if common_dmesg.GetDefaultLogPoller() == nil {
+		return nil, nil
+	}
+	select {
+	case <-common_dmesg.GetDefaultLogPoller().WaitStart():
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	}
+
+	logItems, err := common_dmesg.GetDefaultLogPoller().Find(since)
+	if err != nil {
+		return nil, err
+	}
+
+	events := make([]components.Event, 0)
+	for _, logItem := range logItems {
+		if logItem.Error != nil {
+			continue
+		}
+		if logItem.Matched == nil {
+			continue
+		}
+		if logItem.Matched.Name != common_dmesg.EventPowerSupplyInsufficientPowerOnPCIe {
+			continue
+		}
+
+		events = append(events, components.Event{
+			Time:    logItem.Time,
+			Name:    common_dmesg.EventPowerSupplyInsufficientPowerOnPCIe,
+			Type:    components.EventTypeCritical,
```

Comment:
for all the newly introduced event, the level should be a warning. after the first release, we monitor it and if we feel good we promote it to critical in the next release.